x-pack/plugins/stack_connectors/public/connector_types/slack_api/slack_connectors.tsx

```diff
@@ -33,15 +42,123 @@ const getSecretsFormSchema = (docLinks: DocLinksStart): SecretsFieldSchema[] =>
+      isDisabled,
+      isLoading,
+      noSuggestions: false,
+      options,
+    },
+  },
+];
+
 const SlackActionFields: React.FC<ActionConnectorFieldsProps> = ({ readOnly, isEdit }) => {
-  const { docLinks } = useKibana().services;
+  const {
+    docLinks,
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const form = useFormContext();
+  const { setFieldValue } = form;
+  const [formData] = useFormData({ form });
+  const [authToken, setAuthToken] = useState('');
+  const [channels, setChannels] = useState<EuiComboBoxOptionOption[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const abortCtrlRef = useRef(new AbortController());
+  const isMounted = useRef(false);
+
+  async function fetchChannels(newAuthToken: string) {
+    setIsLoading(true);
+    setChannels([]);
+    setFieldValue('config.allowedChannels', []);
+    isMounted.current = true;
+    abortCtrlRef.current.abort();
+    abortCtrlRef.current = new AbortController();
+
+    try {
+      const res = await http.post<GetChannelsResponse>(
+        `${INTERNAL_BASE_STACK_CONNECTORS_API_PATH}/_slack_api/channels`,
+        {
+          body: JSON.stringify({
+            authToken: newAuthToken,
+          }),
+          signal: abortCtrlRef.current.signal,
+        }
+      );
+
+      setIsLoading(false);
+      if (isMounted.current && res.ok) {
+        setChannels(
+          (res?.channels ?? []).map((channel: ChannelsResponse) => ({
+            label: channel.name,
+          }))
+        );
+        toasts.addSuccess(i18n.SUCCESS_FETCH_CHANNELS);
+      }
+
+      return res;
+    } catch (error) {
+      if (isMounted.current) {
+        setIsLoading(false);
+        setChannels([]);
+        if (error.name !== 'AbortError') {
+          toasts.addDanger(error.body?.message ?? i18n.ERROR_FETCH_CHANNELS);
+        }
+      }
+    }
+  }
+
+  const configFormSchemaAfterSecrets = useMemo(
+    () => getConfigFormSchemaAfterSecrets(channels, isLoading, channels.length === 0),
+    [channels, isLoading]
+  );
+
+  const debounceSetToken = debounce(setAuthToken, INPUT_TIMEOUT);
+  useEffect(() => {
+    if (formData.secrets && formData.secrets.token !== authToken) {
+      debounceSetToken(formData.secrets.token);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [formData.secrets]);
+
+  useEffect(() => {
+    if (isMounted.current && authToken && authToken.length > 0) {
+      fetchChannels(authToken);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [authToken]);
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+      abortCtrlRef.current.abort();
+    };
+  }, []);
 
   return (
     <SimpleConnectorForm
       isEdit={isEdit}
       readOnly={readOnly}
       configFormSchema={[]}
       secretsFormSchema={getSecretsFormSchema(docLinks)}
+      configFormSchemaAfterSecrets={configFormSchemaAfterSecrets}
```

Comment:
I get the reasoning behind it (given also time constraints) but I think the slack connector should have its own customized component and not use the `SimpleConnectorForm`. The reason is not the position of the configs after the secrets but mostly that the secrets are connected with these configs. We should disable the channels combo box until the user has filled out the secrets and provided a helpful message why they are disabled and what the user should do to be able to config the allowed channels. In Swimlane (more complex), we use steps where in step 1 you put the URL and the password and in step 2 you configure the rest of the options (I am not proposing this complex UI for the slack connector).

Reply:
Ok, I just see what you did with `configFormSchemaAfterSecrets`. You connected them with `getConfigFormSchemaAfterSecrets`. Make sense.